src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java

```diff
@@ -110,9 +111,21 @@ public MaxwellKafkaProducer(MaxwellContext context, Properties kafkaProperties,
 		this.inflightMessages = new InflightMessageList();
 	}
 
+	private Integer getNumPartitions(String topic) {
+		try {
+			return this.kafka.partitionsFor(topic).size(); //returns 1 for new topics
+		} catch(TimeoutException e) {
+			LOGGER.error("Topic '" + topic + "' name does not exist. Exception: " + e.getLocalizedMessage());
+			throw e;
```

Comment:
hmm... i'm more curious why KafkaProducer.java doesn't have a listTopics() function, but KafkaConsumer.java does. it seems like a better solution to add the missing functionality there. KafkaProducer has a cluster object which it uses for the partitionsFor() call. this also has a topics() call. it seems like the producer could use that call to return a list of topics.
the big downside is that it would be proposing a change to kafka which would be a long poll, but, i think, much cleaner in the long run.
code refs:
https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java#L1326
https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java#L629
https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java#L755
https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/Cluster.java#L238